src/MTGCore/Services/SearchFilterMapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using MTGCore.Services.Exceptions;
+using MTGCore.ViewModels;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace MTGCore.Services
+{
+    public class SearchFilterMapper : ISearchFilterMapper
+    {
+        private Dictionary<string, string> filterParameters = new Dictionary<string, string>();
```

Comment:
I would probably change this to a QueryBuilder as it will make your life easier (it will handle concatenating the query string values in the correct way).

You could use an extension so you wouldn't have to do a null check too.

![image](https://user-images.githubusercontent.com/17239606/119727432-2a466780-be6a-11eb-855c-e214d86a9099.png)

The query string can then be obtained by doing `.ToQueryString().Value`.